### DISCUSSION TITLE
How to combine schema definitions scattered across nested folders/files into one schema?

### DISCUSSION THREAD

**Discussion by xoldd** [hayes/pothos#273] [Q&A] [answered]
Found the solution. Please delete this discussion. Sorry for the inconvenience.

**Comment by hayes** [comment, marked as answer]
Hey, sorry it took me a bit to get around to answering this.

Generally the easiest way to do this is to create a structure like the following:

`/builder.ts`
```ts
import SchemaBuilder from '@pothos/core';
export const builder = new SchemaBuilder({});
```
`/types/{typeName}.ts` (Or however you want to split up your files
```ts
import { builder } from '../builder';

// exported so it can be imported/referenced by other types, but not required by Pothos
export const User = builder.objectRef<UserShape>('User')

User.implement({ fields: t => ({...}) });
```
`/schema.ts`
```ts
import { builder } from './builder';
// import all your files that define types here
import './types/user';
import './types/posts';

export const schema = builder.toSchema({});
```

Now when you import your schema, that file imports each of the files that defines part of your schema, causing those definitions to be added to the builder before calling `builder.toSchema({})`

**Reply by hayes** [reply]
Added this example (will add a lot more in the future) that has some examples of how to build a schema thats spread across multiple files: https://github.com/hayes/pothos/tree/main/examples/complex-app
